artifacts/definitions/Linux/Events/NewFiles.yaml

```diff
@@ -0,0 +1,36 @@
+name: Linux.Events.NewFiles
+description: |
+  This artifact collects Every new file in monitored 
+  directory from the Linux kernel.
+
+precondition: SELECT OS From info() where OS = 'linux'
+
+type: CLIENT_EVENT
+
+required_permissions:
+  - EXECVE
+
+sources:
+  - query: |
+
+     LET new_file_rules = ("-w /etc/init.d -p wa -k vrr_etc_update", "-w /etc/rc.local -p wa -k vrr_etc_update", "-w /etc/modules -p wa -k vrr_etc_update", "-w /etc/cron.allow -p wa -k vrr_etc_update", "-w /etc/cron.deny -p wa -k vrr_etc_updae", "-w /etc/cron.d/ -p wa -k vrr_etc_update", "-w /etc/cron.daily/ -p wa -k vrr_etc_update", "-w /etc/cron.hourly/ -p wa -k vrr_etc_update", "-w /etc/cron.monthly/ -p wa -k vrr_etc_update", "-w /etc/cron.weekly/ -p wa -k vrr_etc_update", "-w /etc/crontab -p wa -k vrr_etc_update", "-w /var/spool/cron -p wa -k vrr_etc_update")
```

Comment:
I see that in those rules there are the crontab directories. So when a crontab is entry is changed/removed the following output is produced: 

> 
> [INFO] 2022-06-10T11:00:39Z hash /var/spool/cron/crontabs/tmp.WStmBd: lstat /var/spool/cron/crontabs/tmp.WStmBd: no such file or directory 
> [INFO] 2022-06-10T11:00:39Z hash /var/spool/cron: Only regular files supported 
> [INFO] 2022-06-10T11:00:39Z hash /var/spool/cron/crontabs/tmp.WStmBd: lstat /var/spool/cron/crontabs/tmp.WStmBd: no such file or directory 
> [
>  {
>   "Time": "2022-06-10T11:00:38.726Z",
>   "UserId": 0,
>   "User": "root",
>   "State": "success",
>   "FileName": "crontabs/tmp.WStmBd",
>   "FilePath": "/var/spool/cron/crontabs/tmp.WStmBd",
>   "Hash_Sha256": null,
>   "Hash_Sha1": null
>  },
>  {
>   "Time": "2022-06-10T11:00:38.73Z",
>   "UserId": 0,
>   "User": "root",
>   "State": "success",
>   "FileName": "",
>   "FilePath": "/var/spool/cron/",
>   "Hash_Sha256": null,
>   "Hash_Sha1": null
>  },
>  {
>   "Time": "2022-06-10T11:00:38.73Z",
>   "UserId": 0,
>   "User": "root",
>   "State": "success",
>   "FileName": "crontabs/tmp.WStmBd",
>   "FilePath": "/var/spool/cron/crontabs/tmp.WStmBd",
>   "Hash_Sha256": null,
>   "Hash_Sha1": null
>  }
> ]

I think that's suboptimal because audit can't distinguish between the actual user spool file and the temporary files created by 'crontab -e'. This is being handled in the cronsnoop plugin I'm working on, so how do we want to deal with this - here in audit or in the cronsnoop plugin?

Reply:
If the cronsnoop plugin is smarter, let's use that.